mayer1a/MEOWLSServer
# T-1 Add auth, validation, refactor

- Add Errors, validation & authenticator:
  - Add CustomErrorMiddleware
  - Add CustomTokenAuthorization
  - Add Authenticator
  - Add ValidationFailure
  - Add CustomError
  - Add ModelCustomTokenAuthenticatable
  - Add Dummy response
- Add Sequence+Async extenstion
- Add Routes+allPrintable extension
- Refactor swiftSettings in Package.swift
- Refactor Dockerfile
- Add custom working directory to main app scheme
- Add new User DB model
- Add new UserRole DB type
- Add new Token DB model
- Add new Gender DB type
- Add UserRole
- Rename User, Token and Gender migrations
- Add User.Create, .Update, .Public DTO models
- Add UserValidatable protocol
- Add User.Create+Validation extension
- Add User.Update+Validation extension
- Add using CustomErrorMiddleware
- Refactor configure func: add db connection, replace migration to separate func
- Refactor UserController for new User models
- Add 2 test static images
- Replace configure func to Configuration struct
- Replace registerRoutes func to RegisterRoutes struct
- Remove legacy registerPOSTRoutes
- Add «health-check» path for render.com
